sys/events.js

```diff
@@ -85,6 +85,26 @@ EventService.prototype._emit = function(hyper, req) {
                 return event;
             })
             .filter(function(event) { return !!event; });
+
+            // Change-propagation will set up the x-rerender-reason header, indicating
+            // the event which caused the rerender. In case RESTBase is about to emit
+            // the same event, it will cause a rerender loop. So, log an error and skip
+            // an event.
+            var reRenderReason = req.headers && req.headers['x-rerender-reason']
+                || hyper._rootReq && hyper._rootReq['x-rerender-reason'];
+            if (reRenderReason) {
+                events = events.filter(function(event) {
+                    if (reRenderReason.replace(/^https?:/, '') === event.meta.uri.replace(/^https?:/, '')) {
```

Comment:
> It would be good to include the topic as well. Perhaps pre-process the trigger / reason header into an object, and then match on that?

I agree. The combo is what makes the event unique.

> Also, why do we need to replace https:/ here? I thought our internal events are all proto-relative, is that not the case?

I'm reading this as making sure it's proto-relative and as such I think it makes for more solid code.`